### DISCUSSION TITLE
I'm new to zwave and zwavejs2mqtt and a little confused. help please!

### DISCUSSION THREAD

**Discussion by lordmundi** [zwave-js/zwave-js-ui#2045] [Q&A]
I've bought my first zwave stick (the Zooz S2 stick 700). I am running home assistant and trying to set things up with everything running within HA (I don't have a separate docker host machine or anything).

I did my best to follow the instructions on the zwavejs2mqtt documentation, and the home assistant page on ZwaveJS. Obviously there is some relationship between these packages but I don't know what the final end state of everything should be. For example:

Should both integrations be installed on Home Assistant?
If so, should they both be running? or just one or the other?
If they are both running, they both have a configuration line to specify the Zwave device serial interface (one in the integration config, and for this project, in the GUI under the settings). Do both of them run and access the serial device at the same time?

Thanks for any guidance... I've got things installed but I can't quite tell what the final configuration of these two integrations should be.

**Comment by lordmundi** [comment]
ok, i'm learning a little more - apparently there is a zwaveJS integration and a zwaveJS add-on... and I think those are different things. So I guess i need to extend my question above to be more specific about the add-on and whether that should be installed or running.

**Reply by robertsLando** [reply]
@lordmundi You have two choices, or use only the zwavejs integration or use zwavejs integration paired with zwavejs2mqtt (by enabling zwavejs server option). Zwavejs2mqtt is mostly used as control panel as it provides more powerful features to manage your nodes

**Reply by lordmundi** [reply]
@robertsLando yes, I understand that much. But I'm trying to understand if I should have the zwaveJS addon (which I think is different than the integration) installed and running. I'm not entirely familiar with the difference between the add-on and the integration.

**Reply by robertsLando** [reply]
I don't use HASS, BTW if with the addon you mean the zwavejs 2 mqtt addon the answer is the same I wrote in the post above, you can use the integration even without the addon, the addon gives you more control of your network.

**Reply by lordmundi** [reply]
no i mean the zwave JS add-on. See attached screenshots. One shows the Zwave JS add-on and another shows the Z-wave JS integration.  The add-on has a configuration that wants the path to the serial device and such so it sounds like it will be accessing the z-wave stick.

![image](https://user-images.githubusercontent.com/363018/145838081-c2ff2862-735a-4dad-a34c-543a99e5dfbd.png)

**Reply by robertsLando** [reply]
That **addon is zwavejs2mqtt** so the answer is:

> You have two choices, or use only the zwavejs integration or use zwavejs integration paired with zwavejs2mqtt (by enabling zwavejs server option). Zwavejs2mqtt is mostly used as control panel as it provides more powerful features to manage your nodes

**Reply by lordmundi** [reply]
![image](https://user-images.githubusercontent.com/363018/145839674-5c26b099-44f6-4152-a4d7-e375d0fb190c.png)

**Reply by robertsLando** [reply]
@lordmundi That is the zwavejs integration, the one on its right is the zwavejs2mqtt addon. The integration is used to create entities on Hass side and don't need zwavejs2mqtt running. BTW you can install both and configure the zwavejs server that communicates with the zwavejs integration from zwavejs2mqtt settings by enabling zwavejs server option and configuring there the zwave connection

**Reply by robertsLando** [reply]
https://zwave-js.github.io/zwavejs2mqtt/#/homeassistant/homeassistant-official

**Reply by lordmundi** [reply]
i've read through all of that documentation. i can't tell if i'm just stupid or what. I still don't follow exactly on what I should have running. You are saying that is the ZwaveJS integration and not the add-on, but I also see a ZwaveJS integration under the integration and the configuration is different. The configuration for the ZwaveJS add-on wants the serial path to the device, just like the configuration page on the zwavejs2mqtt GUI config page does.  Right now I have the ZwaveJS add-on stopped and things still seem to be working ok, so something tells me that add-on is not the same as the integration as you said. In fact, I'm almost certain it isn't because as I type this I uninstalled the ZwaveJS add-on (the thing circled above) and the integration is still there on the integrations page. So, I hope you don't take offense, but I don't think your statement above is correct.

**Reply by robertsLando** [reply]
@lordmundi As I said I don't use HASS so I'm tring to guess what you are asking me, AFAIK HASS has a zwavejs integration and a zwavejs2mqtt addon, the zwavejs integration works out of the box and also gives a minimal interface to manage your nodes, so if you have the integration installed maybe that addon is used for the minimal interface (I thought that the interface was embedded with the integration but I may be wrong here)

**Reply by blhoward2** [reply]
There are two addons and the integration. You need one of the addons and the integration.

**Reply by lordmundi** [reply]
@blhoward2 ahh ok... that helps a bit. So as part of this I uninstalled the ZwaveJS add-on and just kept the zwavejs2mqtt add-on. And I still see the ZwaveJS integration... so hopefully that is a good final state to be in.  Thanks for the message!
